CHANGELOG.md

```diff
@@ -7,6 +7,7 @@
 ### Fixed
 * <How do the end-user experience this issue? what was the impact?> ([#????](https://github.com/realm/realm-core/issues/????), since v?.?.?)
 * UserIdentity metadata table grows indefinitely. ([#5152](https://github.com/realm/realm-core/issues/5152), since v10.0.0)
+* Add `IncompatibleHistories` case when translating file exceptions (since v6.0.0).
```

Comment:
Isn't this more an internal change?

Reply:
Yeah more of an internal change. I'll move it there.
And correct on the SDKs.

Reply:
Well for the SDKs, it depends on how they are translating core exceptions I suppose. Previously an error like this would come up as Kind::AccessError, now it would come as Kind::BadHistoryError.

Reply:
It's arguably a bug fix. We used to throw BadHistoryError in this case, but then that broke when core started throwing the correct exception and it turned out that we didn't have any tests which verified that we got the correct error message for this specific case.